orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java

```diff
@@ -55,15 +56,28 @@ public TaskResult execute(@Nonnull Stage stage) {
     GoogleCloudBuildStageDefinition stageDefinition =
         stage.mapTo(GoogleCloudBuildStageDefinition.class);
 
-    Map<String, Object> buildDefinition;
-    if (stageDefinition.getBuildDefinitionSource() != null
-        && stageDefinition.getBuildDefinitionSource().equals("artifact")) {
-      buildDefinition = getBuildDefinitionFromArtifact(stage, stageDefinition);
-    } else {
-      buildDefinition = stageDefinition.getBuildDefinition();
+    GoogleCloudBuild result;
+    String source = Optional.ofNullable(stageDefinition.getBuildDefinitionSource()).orElse("");
+    switch (source) {
```

Comment:
One thought---now that we're adding to the values that can be in `source`, I wonder if it would make sense to change it to an `enum` in `GoogleCloudBuildStageDefinition` (taking `STAGE`, `ARTIFACT`, or `TRIGGER`).  Then the deserialization logic can handle setting it to the enum value (defaulting to `STAGE` if `null` or if we get a `IllegalArgumentException`).  Then consumers of the class can deal only with the enum value and we avoid switches on Strings.

Reply:
Make sense... will implement.